api/turing/api/ensembling_job_api.go

```diff
@@ -0,0 +1,94 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	mlp "github.com/gojek/mlp/client"
+	"github.com/gojek/turing/api/turing/models"
+	"github.com/gojek/turing/api/turing/service"
+)
+
+// EnsemblingJobController is the HTTP controller that handles the orchestration of ensembling jobs.
+type EnsemblingJobController struct {
+	BaseController
+}
+
+// Create is a HTTP Post Method that creates an Ensembling job.
+// This method will only return the acceptance/rejection of the job rather than the actual result of the job.
+func (c EnsemblingJobController) Create(
+	r *http.Request,
+	vars RequestVars,
+	body interface{},
+) *Response {
+	var errResp *Response
+	var project *mlp.Project
+	if project, errResp = c.getProjectFromRequestVars(vars); errResp != nil {
+		return errResp
+	}
+
+	// Check is done in api.handlers
+	request, _ := body.(*models.EnsemblingJob)
+	request.ProjectID = models.ID(project.Id)
+
+	// Check if ensembler exists
+	ensembler, err := c.EnsemblersService.FindByID(
+		request.EnsemblerID,
+		service.EnsemblersFindByIDOptions{
+			ProjectID: &request.ProjectID,
+		},
+	)
+	if err != nil {
+		return NotFound("ensembler not found", err.Error())
+	}
+
+	// Ensembler URI will be a local directory
+	// Dockerfile will build copy the artifact into the local directory.
+	// See engines/batch-ensembler/app.Dockerfile
+	artifactFolderName, err := getEnsemblerFolderName(ensembler)
```

Comment:
This should be done at the service layer, not controller layer. Controller should only be responsible for (de)serialization and validation of the user provided data

Reply:
Ok, I was thinking the service was more of a facade pattern to do all connections to the database. Will move all logic into service instead.

Reply:
No, it's not. There are other services, that do more than interacting with DB, for instance: [outer_deployment_service.go](https://github.com/gojek/turing/blob/main/api/turing/service/router_deployment_service.go) or [crypto_service.go](https://github.com/gojek/turing/blob/main/api/turing/service/crypto_service.go)

Reply:
Yah I noticed this as well.